src/components/Applications.jsx

```diff
@@ -36,10 +36,16 @@ const LoadingAppTiles = memo(({ num }) => {
 })
 LoadingAppTiles.displayName = LoadingAppTiles
 
-export const Applications = () => {
+export const Applications = ({ onLoad }) => {
   const showLogout = !!flag('home.mainlist.show-logout')
   const shortcuts = useHomeShortcuts()
-  const { data, fetchStatus } = useQuery(appsConn.query, appsConn)
+  const { data, fetchStatus, lastFetch } = useQuery(appsConn.query, appsConn)
+  const didLoadOnce = useRef(false)
+
+  useEffect(() => {
+    if (fetchStatus === 'loading') didLoadOnce.current = true
+    if (didLoadOnce && fetchStatus === 'loaded' && lastFetch) onLoad()
```

Comment:
I don't know if it works tbh. 

I'm not sure that the first render of this component, we'll have the `fetchStatus === loading`. 

I think it's more something like: 
1st render: fetchStatus === 'loaded' && lastFetch 
2nd render: fetchStatus === 'loading' && lastFetch 

So something like: 
const hasBeenAlreadyLoaded = useRef(false)
```
if(fetchStatus === 'loaded' && lastFetch && !hasBeenAlreadyLoaded) {
hasBeenAlreadyLoaded.current = true
onLoad()
}
```